transport/noop_mapper.go

```diff
@@ -0,0 +1,65 @@
+package transport
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ThreeDotsLabs/esja/stream"
+)
+
+// NoOpMapper implements an interface of transport.Mapper
+// The mapper will use provided original stream events as transport models.
+type NoOpMapper[T any] struct {
+	supported map[stream.EventName]stream.Event[T]
+}
+
+// NewNoOpMapper returns a new instance of NoOpMapper.
+func NewNoOpMapper[T any](
+	supportedEvents []stream.Event[T],
+) *NoOpMapper[T] {
+	supported := make(map[stream.EventName]stream.Event[T])
+	for _, e := range supportedEvents {
+		supported[e.EventName()] = e
+	}
+
+	return &NoOpMapper[T]{
+		supported: supported,
+	}
+}
+
+// RegisterEvent registers new supported stream event.
+func (m *NoOpMapper[T]) RegisterEvent(e stream.Event[T]) {
+	m.supported[e.EventName()] = e
+}
```

Comment:
Given that the constructor accepts a list of supported events, this `register` behavior feels like a code smell to me. Especially since it just overrides whatever was there.

Do you see any use-case for using `RegisterEvent` on an already existing mapper?

The mappers' methods could also use a value receiver if not for this method.

Reply:
Makes sense! I think we can remove it.

Reply:
As this method was the only reason for Mapper mutability I removed pointer receivers as well. https://github.com/ThreeDotsLabs/esja/pull/18/commits/6da74f96db64276e035dbd492c618a6f2b3b6396